sdk/core/azure-core/inc/azure/core/credentials.hpp

```diff
@@ -36,6 +35,19 @@ namespace Azure { namespace Core {
     DateTime ExpiresOn;
   };
 
+  namespace Http {
+    class TransportPolicy;
+  }
```

Comment:
Why are we dong this?

Reply:
It is a forward declaration to let the compiler know that whenever it sees `Http::TransportPolicy`, it is a class named `TransportPolicy` inside the `Http` namespace. We can't include `policy.hpp` here, because `policy.hpp` needs `credentials.hpp`, since `BearerTokenAuthenticationPolicy` does contain `GetTokenOptions` by value a a member, and that requires the full `GetTokenOptions` declaration to be known to the compiler, not just the mere knowledge that it is a class - it needs to know all the members, because it needs to know the `sizeof(GetTokenOptions)` in order for it to be a class member contained by value. `GetTokenOptions`, on the other hand, is fine just knowing that `Http::TransportPolicy`, it is a class named `TransportPolicy` inside the `Http` namespace, because, yes, it is used as a data member of `GetTokenOptions`, but it is contained as a pointer, and for that, you don't need the `sizeof(Http::TransportPolicy)` - compiler is using a `sizeof(pointer)`, and all pointers (to data types) are of the same size, be it `int*` or `Http::TransportPolicy*`.